### DISCUSSION TITLE
Transparent Mode on Windows!

### DISCUSSION THREAD

**Discussion by nguyenhoangdai** [mitmproxy/mitmproxy#5540] [Q&A]
Hi everyone,
I need some help with setting up Transparent Mode and clarifying things. I appreciate any help to address these following questions. 
Information: 
Mitmproxy: 8.1.1
Python: 3.10.6
OpenSSL: OpenSSL 3.0.5 5 Jul 2022
Platform: Windows-10-10.0.19044-SP0

1 - I installed MITMPROXY and used Transparent Mode. I enabled IP Forwarding on Windows10, disables ICMP redirects and tried out Transparent mode immediately. It was working perfectly but I thought 80, 443 traffic was supposed to be redirected to 8080 in order to be picked up by mitmproxy? Is that because after I enabled IP forwarding , it did that job for me or mitmproxy was able to pick up without redirecting port. Any explanation on this part would be really helpful.

2 - Mitmproxy listens on port 8080 by default. However, I have an application also listens on 8080. I want to re-configure mitmrpoxy to listen on 8081 using --listen-port 8081 options. However, How do I set up traffics to redirect to 8081 for mitmproxy to pick up on Windows10? Noted that I tried port forwarding but if I'm not mistaken, port forwarding is at NAT and if it already reaches NAT, which means NAT is already applied and it won't work. 

@mhils Would you mind helping me on these 2 questions? Thank you in advance!

**Comment by yangislet** [comment]
i get same trouble to setup transparent proxy in wondows 10 with mitmproxy v5.3 version .   that is really upset
